packages/monitor-v2/src/llm-bot/OptimisticOracleV2Client.ts

```diff
@@ -0,0 +1,113 @@
+import { Provider } from "@ethersproject/abstract-provider";
+import { paginatedEventQuery } from "@uma/common";
+import { RequestPriceEvent } from "@uma/contracts-frontend/dist/typechain/core/ethers/OptimisticOracleV2";
+import { OptimisticOracleV2Ethers } from "@uma/contracts-node";
+import { Event, EventFilter, ethers } from "ethers";
+import { blockDefaults } from "../utils/constants";
+import { getContractInstanceWithProvider, tryHexToUtf8String } from "../utils/contracts";
+import {
+  BlockRange,
+  OptimisticOracleClient,
+  OptimisticOracleRequest,
+  OptimisticOracleType,
+  calculateRequestId,
+} from "./common";
+
+export class OptimisticOracleClientV2 extends OptimisticOracleClient<OptimisticOracleRequest> {
+  constructor(
+    _provider: Provider,
+    _requests: Map<string, OptimisticOracleRequest> = new Map<string, OptimisticOracleRequest>(),
+    _fetchedBlockRanges?: BlockRange[]
```

Comment:
Why do we need an array of ranges? I would expect every client to cover exactly one contiguous range, no? Would we ever need to have odd, non contiguous ranges like `[[1,2][13,17],[189,456]]`?

Reply:
You are right! This was inherited from when we allowed having gaps. Doesn't make sense anymore. Fixed